src/chainrules.jl

```diff
@@ -193,3 +193,66 @@ ChainRulesCore.@scalar_rule(
 ChainRulesCore.@scalar_rule(expinti(x), exp(x) / x)
 ChainRulesCore.@scalar_rule(sinint(x), sinc(invπ * x))
 ChainRulesCore.@scalar_rule(cosint(x), cos(x) / x)
+
+# non-holomorphic functions
+function ChainRulesCore.frule((_, _, _), ::typeof(besselix), ν::Number, x::Number)
+    Ω = besselix(ν, x)
+    ΔΩ = ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO)
+    return Ω, ΔΩ
+end
+function ChainRulesCore.rrule(::typeof(besselix), ν::Number, x::Number)
+    Ω = besselix(ν, x)
+    project_x = ChainRulesCore.ProjectTo(x)
+    function besselix_pullback(ΔΩ)
+        ν̄ = ChainRulesCore.@not_implemented(BESSEL_ORDER_INFO)
+        a = (besselix(ν - 1, x) + besselix(ν + 1, x)) / 2
+        b = - sign(real(x)) * Ω
+        x̄ = project_x(conj(a) * ΔΩ + real(conj(b) * ΔΩ))
+        return ChainRulesCore.NoTangent(), ν̄, x̄
+    end
+    return Ω, besselix_pullback
+end
+
+function ChainRulesCore.frule((_, _, _), ::typeof(besseljx), ν::Number, x::Number)
```

Comment:
Since everything should be identical for `besseljx` and `besselyx`, can you declare them the same with a loop and an `@eval`?

Reply:
I could but so far the style in this file is to implement everything explicitly (eg. also derivatives `besselj` and `bessely` are implemented separately) and so I sticked to it.

Reply:
Makes sense.